sopel/plugins/capabilities.py

```diff
@@ -129,11 +228,40 @@ def request_available(
         :param bot: the bot instance used to send capability requests
         :param available_capabilities: available capabilities
 
-        This sends ``CAP REQ`` command for requests that can be made, i.e.
+        This sends ``CAP REQ`` commands for requests that can be made, i.e.
         all the requested capabilities (with or without prefix) must be
         available for Sopel to send the request.
 
-        Requests made are stored as requested, other are ignored.
+        Requests made are stored as requested; others are ignored::
+
+            >>> # manager.register => cap1 and cap2, not cap3
+            >>> manger.request_available(bot, ('cap1', 'cap3'))
+            >>> manger.is_requested(('cap1',))
+            True
+            >>> manger.is_requested(('cap2',))
+            False
+            >>> manger.is_requested(('cap3',))
+            False
+
+        .. important::
+
+            The capability request ``('cap1', '-cap2')`` means "enable cap1 and
+            disable cap2", and the request will be acknowledged or denied at
+            once. If the server doesn't advertise any of these capabilities,
+            the client **should not** send a request.
+
+            As a result, Sopel will send request to enable or disable a
+            capability only if it is advertised first. As a result, if a
+            request is never made, its callback will never be called, because
+            there won't be any ACK/NAK message from the server for them.
+
+            Plugin author should not use the request's callback to activate or
+            deactivate features, and instead check the bot's capabilities every
+            time they need to.
+
+            The only exception to that is when the plugin needs to perform an
+            operation while negociating the capabilities (such as SASL auth).
```

Comment:
```suggestion
            As a result, Sopel will send a request to enable or disable
            a capability only if it is advertised first. If a request is
            never made, its callback will never be called, because there
            won't be any related ACK/NAK message from the server.

            Plugin authors should not use the request's callback to activate or
            deactivate features, and instead check the bot's capabilities every
            time they need to.

            The only exception to that is when the plugin needs to perform an
            operation while negotiating the capabilities (such as SASL auth).
```

(Rewrap is sloppy, but good enough.)